### PR TITLE
Fixes attr-for-selected issue and adds animate-first-selected feature.

### PR DESCRIPTION
**Pull request by zerodevx**
Hi @morethanreal,

Wrt issue #8 , here are the proposed code changes for your reference.

Thanks so much,
Jason

### CODE REVIEW

File: neon-animated-pages.html [hunk 1]
```diff
@@ -81,54 +89,64 @@
...
-        });
-      }
-
+      selectedPage.classList.add('neon-animating');
+      
       if (this.animationConfig.length > 1) {
-        this.playAnimation();
+        //this.playAnimation();
+        // in the event of first load, we want to run animations only after current microtasks are complete (ie element attached)
+        this.async(this.playAnimation);
```
Comment (morethanreal) [inline neon-animated-pages.html]: Can it only async if one of the pages is not attached? Check for `isAttached` maybe?

File: neon-animated-pages.html [hunk 2]
```diff
@@ -55,7 +55,7 @@
   Polymer({
 
     is: 'neon-animated-pages',
-
+
```
Comment (morethanreal) [inline neon-animated-pages.html]: Can you remove white spaces from line endings?

File: neon-animated-pages.html [hunk 3]
```diff
@@ -67,11 +67,19 @@
...
-        type: String
+      
+      /**
+       * If true, the initial page selection will also be animated according to its animation config
+       *
+       * @attribute animate-first-selected
+       * @type boolean
+       * @default false
+       */
+      animateFirstSelected: {
```
Comment (morethanreal) [inline neon-animated-pages.html]: Maybe rename to `animate-initial-selection`?

File: neon-animated-pages.html [hunk 4]
```diff
@@ -67,11 +67,19 @@
...
-
-      _prevSelected: {
-        type: String
+      
+      /**
+       * If true, the initial page selection will also be animated according to its animation config
+       *
+       * @attribute animate-first-selected
+       * @type boolean
+       * @default false
```
Comment (morethanreal) [inline neon-animated-pages.html]: I think hydrolysis is able to parse the property block without these annotations.

**Comment by morethanreal** [comment]
Thanks for your contribution. In addition to the comments, would you mind splitting the animate first selection feature and the fix for attr-for-selected into separate commits? I also committed a fix for https://github.com/PolymerElements/neon-animation/issues/11, so you'll need to rebase. Thank you!

**Comment by zerodevx** [comment]
Thanks for your notes, @morethanreal. I've

1. Renamed feature to `animate-first-selection`;
2. Split into [fix-for-attr-selected] (https://github.com/zerodevx/neon-animation/tree/fix-attr-for-selected) and [add-animate-initial-selection] (https://github.com/zerodevx/neon-animation/tree/add-animate-initial-selection) branches.
3. Removed white-spaces;
4. Added an `isAttached` check so that async is run only if the element is not yet attached;
5. Removed annotations.

I'll submit the PR for `animate-initial-selection` in a bit.

Thanks so much,
Jason

**Comment by morethanreal** [comment]
lgtm

**Pull request merged**